lastseen.py

```diff
@@ -0,0 +1,46 @@
+import re
+from datetime import datetime
+from errbot import BotPlugin, botcmd, re_botcmd
+
+
+USAGE_STR = 'Nick not specified. (usage: !last_seen <nick>)'
+
+
+class LastSeen(BotPlugin):
+
+    def activate(self):
+        super(LastSeen, self).activate()
+        try:
+            if type(self['last_seen']) is not dict:
+                self['last_seen'] = {}
+        except KeyError:
+            self['last_seen'] = {}
+
+    @re_botcmd(pattern=r'^', prefixed=False, flags=re.IGNORECASE)
+    def update_last_seen(self, msg, match):
+        user = str(msg.frm).split('!')[0]
```

Comment:
@JakubNvk I understand where this is coming from (the plugin has been primarily aimed at IRC), I believe it would be better to make this as general as possible.

Namely, the `msg.frm` should be of type `Person`. This should allow us to call its [`nick`](http://errbot.io/en/latest/errbot.backends.base.html#errbot.backends.base.Person.nick) function that may directly replace this code in a general manner.